collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+# Managing Kyma components using Kyma CLI
+
+Kyma cli has been used for cluster management. We should look into how we can introduce support for management of components on the kyma cluster. This would bring in following benefits:
+
+* Enable users who have limited or no kuberenetes knowledge to use kyma.
+* Provide an abstraction on complex kubectl command chains.
+* Could be easily used into CI/CD pipelines where they can have automated way of configuring kyma cluster to have functions deployed via CI/CD pipeline
+
+## Architecture
+
+We should follow the modular approach with each component having its own library. This library can be used inside the kyma cli for acessing various components. The library contains all the implementation. The kyma cli should invoke this library. We can take inspiration from hydroform and implement it in similar way. Currently we have following modules in kyma:
```

Comment:
Nice 👍 Having a sdk would be great and could replace a lot of our logic in tests.

Reply:
The sdk can evolve in time, covering aspects like: management of Kyma components, rapid application development (like Dapr's languages sdk) and of course local/remote development.